DNS: Configure root domains

mrsheepuk opened the pull request:
## Summary

Introduces a specific UI to manage root domains for Kore.


**Which issue(s) this PR resolves**:
Resolves #1297 

## Changes

Please detail any major changes and provide the minimal documentation about how these work or do alter the current behaviour.
Anyone who reads this should understand the intention of this PR without reading the code.

### Additional changes

* API support for filtering on service kind when listing team services.

## Screenshots

![image](https://user-images.githubusercontent.com/7505593/93353719-26226e80-f834-11ea-993b-a52438b144b5.png)

![image](https://user-images.githubusercontent.com/7505593/93353744-30dd0380-f834-11ea-9651-7b4832f98060.png)

## Testing

The following testing has been performed:

* Ensure kore has a set of AWS (EKS) credentials set up
* Open UI, log in as admin
* Select Configure -> DNS
* Add a new domain for kore to manage (e.g. `testytest.kore.appvia.io`)
* Observe that the domain is pending - you should see 'Action required'
* Find the action that is required by clicking on it, follow the instructions to set up the domain.
* After actions are performed, within some time (can take 5-30 minutes depending on DNS timeouts) the state should go to Success.

lewismarshall commented:
@mrsheepuk so looks good and works but would handle closing the draw automatically (as discussed) and then removing the text:
> Please check back later.

Would also remove the text (as also discussed) around:
> For example, add a record such as the following to the authoritative DNS server for labs.kore.appvia.io:
lewis-tests.labs.kore.appvia.io. 300 IN NS
                     ns-195.awsdns-24.com.
                     ns-920.awsdns-51.net.
                     ns-2015.awsdns-59.co.uk.
                     ns-1453.awsdns-53.org.

mrsheepuk commented:
> @mrsheepuk so looks good and works but would handle closing the draw automatically (as discussed) and then removing the text:
> 
> > Please check back later.
> 

Updated this to show the status and auto-update the status in the drawer if you have it open, so it will dynamically change from this:

![image](https://user-images.githubusercontent.com/7505593/93372514-e5ceea80-f84b-11ea-8046-fc674615d6ab.png)

.. to this ..

![image](https://user-images.githubusercontent.com/7505593/93372557-f3847000-f84b-11ea-9754-b8a95902886b.png)

Also tightened the wording and removed the extraneous text as suggested.

lewismarshall reviewed (approved):
LG - all working

davet1985 commented on ui/lib/kore-api/kore-api-resources.js:
```diff
@@ -375,6 +375,35 @@ export default class KoreApiResources {
+  /**
+   * Generates a CRD for a system service (i.e. one hosted by the kore cluster)
+   */
+  generateSystemServiceResource(team, values, serviceName, planName, kindName) {
```
Minor, but could add a test for this to keep the API models fully tested.

davet1985 commented on ui/pages/configure/dns/index.js:
```diff
@@ -0,0 +1,15 @@
+    return (
+      <>
+        <Breadcrumb items={[{ text: 'Configure' }, { text: 'DNS' }]} />
+        <DomainList namespace="kore-admin" />
```
This should probably use the `koreAdminTeamName` from the next config.

davet1985 commented on ui/lib/kore-api/kore-api-resources.js:
```diff
@@ -375,6 +375,35 @@ export default class KoreApiResources {
+      group: 'clusters.compute.kore.appvia.io',
+      kind: 'Cluster',
+      name: 'kore',
+      namespace: 'kore-admin',
```
This should also be using the `koreAdminTeamName` from the next config.
